src/fluree/db/ledger/mutable.clj

```diff
@@ -58,15 +60,38 @@
             (log/warn (str "Flakes purged from block " block))
             true)))
 
+
+(defn- create-block-map
+  "Creates a block-map from a vector of flakes"
+  [db flakes]
+  (go-try
+    (loop [[^Flake flake & r] flakes
+           t-map     {}
+           block-map {}]
+      (if flake
+        (if-let [block (get t-map (.-t flake))]
+          (recur r t-map (update block-map block conj flake))
+          (let [t     (.-t flake)
+                block (<? (time-travel/non-border-t-to-block db t))]
+            (if (get block-map block)
+              (recur r (assoc t-map t block) (update block-map block conj flake))
+              (recur r (assoc t-map t block) (assoc block-map block [flake])))))
+        block-map))))
+
+
 (defn identify-hide-blocks-flakes
   [db {:keys [block hide purge] :as query-map}]
   (go-try (let [[pattern idx] (fdb/get-history-pattern (or hide purge))
-                [block-start block-end] (when block (<? (fdb/resolve-block-range db query-map)))
+                [block-start block-end]
+                  (when block (<? (fdb/resolve-block-range db query-map)))
                 from-t    (if (and block-start (not= 1 block-start))
-                            (dec (:t (<? (time-travel/as-of-block db (dec block-start))))) -1)
+                            (dec (:t (<? (time-travel/as-of-block db (dec block-start)))))
+                            -1)
                 to-t      (if block-end
-                            (:t (<? (time-travel/as-of-block db block-end))) (:t db))
+                            (:t (<? (time-travel/as-of-block db block-end)))
+                            (:t db))
                 flakes    (<? (query-range/time-range db idx = pattern {:from-t from-t :to-t to-t}))
+                _ (log/info "identify-hide-blocks-flakes " {:flakes flakes})
```

Comment:
Since users run at INFO level logging by default, we should either clean this up so what it's indicating is more clear, or change it to DEBUG level.

Reply:
Changed to debug